src/sqlancer/influxdb/InfluxDBProvider.java

```diff
@@ -0,0 +1,112 @@
+package sqlancer.influxdb;
+
+import com.google.auto.service.AutoService;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+import sqlancer.*;
+import sqlancer.common.query.SQLQueryAdapter;
+import sqlancer.common.query.SQLQueryProvider;
+import sqlancer.influxdb.gen.InfluxDBCreateDatabaseGenerator;
+import sqlancer.influxdb.gen.InfluxDBWritePointGenerator;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Properties;
+
+@AutoService(DatabaseProvider.class)
+public class InfluxDBProvider extends SQLProviderAdapter<InfluxDBProvider.InfluxDBGlobalState, InfluxDBOptions> {
+
+    public InfluxDBProvider() {
+        super(InfluxDBGlobalState.class, InfluxDBOptions.class);
+    }
+
+    public enum Action implements AbstractAction<InfluxDBGlobalState> {
+        WRITE_POINT(sqlancer.influxdb.gen.InfluxDBInsertGenerator::getQuery),
```

Comment:
Is it intended that both of this refers to the `InsertGenerator`? I think it would also be good to stick to the existing naming conventions, which use `INSERT` as part of the action name.